### PR TITLE
[14.0][MIG] stock_account_prepare_anglo_saxon_out_lines_hook

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
No changes in core. This module will not be needed in v15 as long as the hook was accepted in master

@ForgeFlow

Edit: the module in v15 is needed. One of the hooks is merged in core but not the other

### CODE REVIEW

**Review by JoanMForgeFlow — APPROVED** [review]
Tested on Local, works fine! 
thanks

**Review by mariadforgeflow — APPROVED** [review]
Code review OK.

File: stock_account_prepare_anglo_saxon_out_lines_hook/hooks.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+                # Filter out lines being not eligible for COGS.
+                # FIRST HOOK STARTS
+                if not line._eligible_for_cogs():
+                    continue
+                # FIRST HOOK ENDS
+
+                # Retrieve accounts needed to generate the COGS.
+                accounts = line.product_id.product_tmpl_id.with_context(
+                    force_company=line.company_id.id
```
Comment (MiquelRForgeFlow) [inline stock_account_prepare_anglo_saxon_out_lines_hook/hooks.py]:
```suggestion
                accounts = line.product_id.product_tmpl_id.with_company(
                    line.company_id
```

File: stock_account_prepare_anglo_saxon_out_lines_hook/hooks.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                )
+                if not credit_expense_account:
+                    if self.type == "out_refund":
+                        credit_expense_account = (
+                            self.journal_id.default_credit_account_id
+                        )
+                    else:  # out_invoice/out_receipt
+                        credit_expense_account = (
+                            self.journal_id.default_debit_account_id
+                        )
```
Comment (MiquelRForgeFlow) [inline stock_account_prepare_anglo_saxon_out_lines_hook/hooks.py]:
```suggestion
```
see https://github.com/odoo/odoo/commit/238a5260d51b35ef32c022348792c8b63b72078e

File: stock_account_prepare_anglo_saxon_out_lines_hook/hooks.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                if not debit_interim_account or not credit_expense_account:
+                    continue
+
+                # Add interim account line.
+                # SECOND HOOK STARTS
+                if not debit_interim_account or not credit_expense_account:
+                    continue
```
Comment (MiquelRForgeFlow) [inline stock_account_prepare_anglo_saxon_out_lines_hook/hooks.py]:
```suggestion
```
it's repeated

**Comment by AaronHForgeFlow** [comment]
@MiquelRForgeFlow I agree with you. I applied your suggestions. Thanks for the help!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by rvalyi** [comment]
Hello, just one question for curiosity, do you use IFRS and the anglo saxon mode in Spain?

**Comment by AaronHForgeFlow** [comment]
> Hello, just one question for curiosity, do you use IFRS and the anglo saxon mode in Spain?

I do this mainly for US companies, I am not sure if there is a company in Spain using that but I don't think so.
Regards :)

**Pull request merged**